perf/Map.js

```diff
@@ -139,4 +139,39 @@ describe('Map', function () {
 
   });
 
+  describe('merge an object', function () {
+
+    [2, 8, 32, 1024].forEach(size => {
+      const obj = {};
+      for (let ii = 0; ii < size; ii++) {
+        obj['x' + ii] = ii;
+      }
+
+      const m1 = Immutable.Map(obj);
+      const m2 = Immutable.Map(obj);
```

Comment:
These two are the same - there's a special path for no-op merges, so you might want to add a separate perf test  where the second map contains different values